File: week3/java/week3_6.java
public class week3_6 {
    public static int solution(int[] nums){
        int answer = 0;
        int check[] = new int[200001];
        int count = 0;

        int i = 0;
        while(count != nums.length/2 && i<nums.length){
            if(check[nums[i]]==0){
                count++;
                answer++;
                check[nums[i]]++;
            }
            i++;
        }

        return answer;
    }
    public static void main(String[] args){
        int[] arr = {3,3,3,2,2,4};
        System.out.println(solution(arr));
    }
}

File: week3/c++/week3_6.cpp
#include <vector>
using namespace std;

int visit[200001];
int solution(vector<int> nums)
{
    int n = nums.size();

    int count = 0;
    int index = 0;
    while (count < n / 2 && index < n)
    {
        if (visit[nums[index]] == 0)
        {
            visit[nums[index]] = 1;
            count++;
        }
        index++;
    }

    return count;
}